plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/TrinoToMySqlWriteChecker.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mysql;
+
+import io.trino.spi.TrinoException;
+
+import java.time.Instant;
+
+import static io.trino.spi.StandardErrorCode.INVALID_ARGUMENTS;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TrinoToMySqlWriteChecker<T>
+{
+    // The range for TIMESTAMP values is '1970-01-01 00:00:01.000000' to '2038-01-19 03:14:07.499999'
+    public static final TrinoToMySqlWriteChecker<Instant> TIMESTAMP = new TrinoToMySqlWriteChecker<>(
+            new TimestampWriteValueChecker(new Range<>(Instant.parse("1970-01-01T00:00:01.000000Z"), Instant.parse("2038-01-19T03:14:07.499999Z"))));
```

Comment:
Why not 1970-01-01T00:00:00.000000Z?

https://en.wikipedia.org/wiki/Unix_time

Unix time is currently defined as the number of seconds which have passed since 00:00:00 UTC on Thursday, 1 January 1970

Reply:
good catch, this should proably be `1970-01-01T00:00:00.000001Z` - i.e. `epoch + min delta`

Reply:
From the [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/datetime.html):

>The TIMESTAMP data type is used for values that contain both date and time parts. TIMESTAMP has a range of '1970-01-01 00:00:01' UTC to '2038-01-19 03:14:07' UTC.

```
mysql> insert into test_unsupported_timestamp values ('1970-01-01 00:00:00');
ERROR 1292 (22007): Incorrect datetime value: '1970-01-01 00:00:00' for column 'data' at row 1

mysql> insert into test_unsupported_timestamp values ('1970-01-01 00:00:01');
Query OK, 1 row affected (0.01 sec)

mysql> select * from test_unsupported_timestampbykpmm32ki;
+---------------------+
| data                |
+---------------------+
| 1970-01-01 00:00:01 |
+---------------------+
```

Reply:
I'm not sure whether this is a problem or not, but I'll mention it 'just in case'.
Back a few years ago I was working with a JSON interface of an HTTP endpoint. I was surprised to see that I could convert a negative long passed in JSON into Javascript date to represent dates before epoch. We used that for birth dates just like described here: https://www.epochconverter.com/programming/mysql (paragraph 'Negative epochs').

Now, as the article reads, mysql has a problem representing timestamps before epoch. However, we have a 'converter' at hand that is translating trino datatype to particular database datatype. And that trino datatype just might support negative values. Currently the test throws an exception if the timestamp is out of range, for instance if it's before epoch. I'm just wondering whether that's the behaviour we want to have. Maybe we somehow want to emulate negatives (ie. before epoch) timestamps.

Treat it as a general comment that can be ignored.

Reply:
@adamjshook please add a code comment about the min value  to make it more obvious